src/main/java/org/ibp/api/java/impl/middleware/location/LocationServiceImpl.java

```diff
@@ -56,4 +61,29 @@ public long countLocations(final String crop, final LocationSearchRequest locati
 		final ModelMapper mapper = LocationMapper.getInstance();
 		return locations.stream().map(o -> mapper.map(o, LocationDto.class)).collect(Collectors.toList());
 	}
+
+	@Override
+	public void deleteLocation(final Integer locationId) {
+		this.locationValidator.validateCanBeDeleted(locationId);
+		this.locationMiddlewareService.deleteLocation(locationId);
+	}
+
+	@Override
+	public Integer createLocation(final LocationRequestDto locationRequestDto) {
```

Comment:
In most of the new services we are returning the created object (DTO) instead of the Id